### ISSUE TITLE
Delete functionality is temporarily disabled

### ISSUE THREAD

**Issue by jkburges** [aodn/aatams#204]
... as otherwise it's pretty easy to inadvertently destroy data (see https://github.com/aodn/aatams/issues/202 for an example)

The main idea I have of how "delete" could be done better, is to do "soft" deletes, i.e. mark records as `deleted`, so that they can be hidden from users, but remain in the db.  This would avoid unexpected cascading deletes, as in #202, and also make it much easier to recover accidentally deleted data - they would just have to be "undeleted".

The main downside is that it would (possibly) complicate some of the logic in the app somewhat.

**Comment by astecken** [comment]
As it is my job to clean up the DB and its contents, e.g. deleting entries that are faulty, double entries etc., I have to be able to delete or soft-delete certain records. It would be great to get this function back or the new soft-delete function activated asap so I can go ahead with things. Thanks!

**Comment by xhoenner** [comment]
Some additional specs, as specified by @jkburges in #257:

"There are two use cases:

1. A user no longer can have or wants access to the DB altogether
2. As above, except at an individual project level

We cannot simply delete the user (or their associated project roles), as the user is still referenced in the system (e.g. as a recoverer), and deleting the user record would result in this associated data also being deleted.

The proper way to model this is to allow users to be `DEACTIVATED`.  This option already exists for `organisations` and `projects`, but not for `people`.

Deactivating someone should disallow their access to the system (or individual projects) and stop email notifications."

**Comment by xhoenner** [comment]
After chatting to @astecken we refined the specs for this issue:
  * sysadmins must be able to deactivate (change status to inactive as for organisations and projects) or "soft delete" users to restrict access to the web app. Soft deletes are critical here as a hard delete of users would result in this associated data also being deleted.
  * sysadmins (along with PIs @astecken ?) must be able to take people off projects. At the moment it's only possible to change access to 'Read Only', suggest having 'No Permission' as another 'Access' option when editing projects. 
  * Hard deletes can be (re-)applied to installations, stations, receivers, tags, and tag releases as doing so will not impact the content of the `detection` table.

What do you think of the above plan @jkburges?

**Comment by jkburges** [comment]
> Hard deletes can be (re-)applied to installations, stations, receivers, tags, and tag releases as doing so will not impact the content of the detection table

This is true, although it *will* affect what is visible in the various detection views.  The raw data (as such) won't be affected though.

In general, a good plan.  Much better than allowing users to trigger cascading deletes - definitely an accident waiting to happen IMO.

**Comment by astecken** [comment]
@xhoenner While this function is not back, could you deal with the list with entries that need to be deleted that I sent you, pls? That would be awesome!

**Comment by xhoenner** [comment]
Here are below the requirements for the soft delete functionality based on the above comments and content issues that @astecken has been unable to fix because of the removal of the delete feature.

1. Implement the soft delete feature wherever the hard delete functionality used to be available.

2. sysadmins must be able to deactivate (change status to inactive as for organisations and projects) or "soft delete" users to restrict access to the web app. Soft deletes are critical here as a hard delete of users would result in this associated data also being deleted. See https://github.com/aodn/aatams/issues/257.

3. sysadmins, along with project PIs, must be able to take people off projects. At the moment it's only possible to change access to 'Read Only', suggest having 'No Permission' as another 'Access' option when editing projects (see https://github.com/aodn/aatams/issues/333).

4. Fix https://github.com/aodn/aatams/issues/306 as part of the soft delete functionality as users can't currently edit or delete their own organisation.
